components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4259,8 +4259,9 @@ public static void validateRequestTenantDomain(String tenantDomainOfApp) throws
             String tenantDomainFromContext = IdentityTenantUtil.getTenantDomainFromContext();
             if (!StringUtils.equals(tenantDomainFromContext, tenantDomainOfApp)) {
                 // This means the tenant domain sent in the request and app's tenant domain do not match.
-                throw new InvalidOAuthClientException("A valid client with the given client_id cannot be found in " +
+                log.error("A valid client with the given client_id cannot be found in " +
```

Comment:
Do need this as a error log ? As it a client error this should be debug log right

Reply:
Changed